warehouse/query-core/src/main/java/datawave/query/jexl/visitors/SatisfactionVisitor.java

```diff
@@ -106,26 +106,20 @@ public Object visit(ASTEQNode node, Object data) {
     }
 
     @Override
-    public Object visit(ASTReference node, Object o) {
+    public Object visit(ASTAndNode and, Object data) {
+        QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(and);
         // Recurse only if not delayed
-        if (!QueryPropertyMarker.findInstance(node).isType(ASTDelayedPredicate.class)) {
-            super.visit(node, o);
+        if (!instance.isType(DELAYED)) {
```

Comment:
if we are checking for delayed and exceeded, then does this not equate to looking for any marker